src/components/NavBar/NavBar.js

```diff
@@ -47,7 +47,10 @@ class NavBar extends React.Component {
         }
 
         var authButton = null;
-        if(this.props.authState === AuthStates.GUEST) {
+        if(
+            JSON.stringify(this.props.permissions)
```

Comment:
You can't directly compare JavaScript objects with the `===` operator. This is the simplest way I found to do so.